src/main/java/com/cloudbees/jenkins/support/impl/AdministrativeMonitorsComponent.java

```diff
@@ -60,62 +62,56 @@
-                        out.println();
-                        out.println("`" + monitor.id + "`");
-                        out.println("--------------");
-                        if (monitor instanceof OldDataMonitor && !suffersFromJENKINS24358()) {
-                            OldDataMonitor odm = (OldDataMonitor) monitor;
-                            for (Map.Entry<Saveable,OldDataMonitor.VersionRange> entry : odm.getData().entrySet()) {
-                                out.println("  * Problematic object: `" + entry.getKey() + "`");
-                                OldDataMonitor.VersionRange value = entry.getValue();
-                                String range = value.toString();
-                                if (!range.isEmpty()) {
-                                    out.println("    - " + range);
-                                }
-                                String extra = value.extra;
-                                if (!StringUtils.isBlank(extra)) {
-                                    out.println("    - " + extra); // TODO could be a multiline stack trace, quote it
-                                }
-                            }
-                        } else {
-                            // No specific content we can show; message.jelly is for HTML only.
-                            out.println("(active and enabled)");
-                        }
-                    }
+                    out.println(SupportUtils.toString(getAdminMonitors(activated)));
                 }
             });
         }
     }
 
-    private static boolean suffersFromJENKINS24358() {
-        VersionNumber version = Jenkins.getVersion();
-        if (version == null) {
-            return false;
-        } else if (version.compareTo(/*JENKINS-19544*/new VersionNumber("1.557")) >=0 && version.compareTo(new VersionNumber(/*JENKINS-24358*/"1.578")) < 0) {
-            if (version.toString().startsWith("1.565.") && version.compareTo(new VersionNumber(/* predicting JENKINS-24358 to be backported here */"1.565.3")) >=0) {
-                return false;
+    private AdminMonitors getAdminMonitors(Map<String, AdministrativeMonitor> activated) {
+        AdminMonitors monitors = new AdminMonitors();
+
+        for (AdministrativeMonitor monitor : activated.values()) {
+
+            if (monitor instanceof OldDataMonitor) {
+                AdminMonitors.OldDataMonitor olddm = new AdminMonitors.OldDataMonitor();
+                OldDataMonitor odm = (OldDataMonitor) monitor;
+                for (Map.Entry<Saveable,OldDataMonitor.VersionRange> entry : odm.getData().entrySet()) {
+
+                    AdminMonitors.OldDataMonitor.OldData data = new AdminMonitors.OldDataMonitor.OldData();
+                    data.setProblematicObject(entry.getKey().toString());
+
+                    OldDataMonitor.VersionRange value = entry.getValue();
+                    data.setRange(value.toString());
+                    data.setExtra(value.extra);
```

Comment:
Missing the `StringUtils.isBlank` call from the original.

I suppose `YAMLFactory` will now take care of the possibility of a multiline stack trace.